app/models/ems_refresh/save_inventory_cinder.rb

```diff
@@ -0,0 +1,129 @@
+#
+# Calling order for EmsCloud
+# - ems
+#   - cloud_volumes
+#   - cloud_volume_backups
+#   - cloud_volume_snapshots
+#   - vms
+#     - storages (link)
+#     - security_groups (link)
+#     - operating_system
+#     - hardware
+#       - disks
+#       - guest_devices
+#     - custom_attributes
+#     - snapshots
+#
+
+module EmsRefresh::SaveInventoryCinder
+  def save_ems_cinder_inventory(ems, hashes, target = nil)
+    target = ems if target.nil?
+    log_header = "EMS: [#{ems.name}], id: [#{ems.id}]"
+
+    # Check if the data coming in reflects a complete removal from the ems
+    if hashes.blank?
+      target.disconnect_inv
+      return
+    end
+
+    _log.info("#{log_header} Saving EMS Inventory...")
+    if debug_trace
+      require 'yaml'
+      _log.debug "#{log_header} hashes:\n#{YAML.dump(hashes)}"
+    end
+
+    child_keys = [
+      :cloud_volumes,
+      :cloud_volume_backups,
+      :cloud_volume_snapshots,
+      :vms,
```

Comment:
vms shouldn't be saved, but I need to change this for the cross linking anyway. I say leave it in for now, and I'll clean it up in my PR.